test/Microsoft.AspNet.Mvc.Razor.Test/DefaultTagHelperActivatorTest.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Razor.Test
+{
+    public class DefaultTagHelperActivatorTest
+    {
+        [Theory]
+        [InlineData("test", 100)]
+        [InlineData(null, -1)]
+        public void Activate_InitializesTagHelpers(string name, int number)
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            services.InitializeTagHelper<TestTagHelper>((h, vc) =>
+            {
+                h.Name = name;
+                h.Number = number;
+                h.ViewDataValue = vc.ViewData["TestData"];
+            });
+            var httpContext = MakeHttpContext(services.BuildServiceProvider());
+            var viewContext = MakeViewContext(httpContext);
+            var viewDataValue = new object();
+            viewContext.ViewData.Add("TestData", viewDataValue);
+            var activator = new DefaultTagHelperActivator();
+            var helper = new TestTagHelper();
+
+            // Act
+            activator.Activate(helper, viewContext);
+
+            // Assert
+            Assert.Equal(name, helper.Name);
+            Assert.Equal(number, helper.Number);
+            Assert.Same(viewDataValue, helper.ViewDataValue);
+        }
+
+        [Fact]
+        public void Activate_InitializesTagHelpersAfterActivatingProperties()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            services.InitializeTagHelper<TestTagHelper>((h, _) => h.ViewContext = MakeViewContext(MakeHttpContext()));
+            var httpContext = MakeHttpContext(services.BuildServiceProvider());
+            var viewContext = MakeViewContext(httpContext);
+            var activator = new DefaultTagHelperActivator();
+            var helper = new TestTagHelper();
+
+            // Act
+            activator.Activate(helper, viewContext);
+
+            // Assert
+            Assert.NotSame(viewContext, helper.ViewContext);
+        }
+
+        private static HttpContext MakeHttpContext(IServiceProvider services = null)
+        {
+            var httpContext = new Mock<DefaultHttpContext>();
+            httpContext.CallBase = true;
+            if (services != null)
+            {
+                httpContext.Setup(c => c.RequestServices).Returns(services);
```

Comment:
Plus both the things that call this method always pass in services - just drop the null check and make it non-optional.